### PR TITLE
Fix backend selector in IPython notebook.

### LINKED ISSUE #432: Backend selection box is empty in IPython notebook.

### PR DESCRIPTION
**Pull request by jgosmann**
Fixes #432.

### CODE REVIEW

**Comment by mundya** [comment]
This works for Steve Furber who'd had the problem of the backends disappearing in the normal GUI after running once.  Is there any plan to merge this in the near future?

**Comment by tcstewar** [comment]
> Is there any plan to merge this in the near future?

Ah, thank for the reminder -- I'll review this and merge today if it all looks good.

File: nengo_gui/exec_env.py [hunk 1]
```diff
@@ -94,12 +106,8 @@ def __enter__(self):
...
 
         if not self.allow_sim:
-            for name in known_modules:
-                try:
-                    mod = importlib.import_module(name)
-                except:
-                    continue
-                found_modules.append(name)
+            discover_backends()
+            for mod in discover_backends().values():
```
Comment (tcstewar) [inline nengo_gui/exec_env.py]: Should ```discover_backends``` be called twice here?

Comment (jgosmann) [inline nengo_gui/exec_env.py]: No, it should not.

**Comment by tcstewar** [comment]
Looks great to me... thanks for fixing this, @jgosmann !

**Comment by mundya** [comment]
Thank you! :)

**Pull request merged**